### PR TITLE
Make storage upload/download have no chunk size by default.

### LINKED ISSUE #546: resumable uploads should try to send the file in one chunk by default
Sending data in one chunk is the recommended approach, documented in a not very clear way [here](https://cloud.google.com/storage/docs/json_api/v1/how-tos/upload#resumable) in the “Uploading the file in chunks” section.

We should allow the user to override the chunk size, making sure that the library ends up using a multiple of 256 KB, as expected from the API (except for the final chunk that completes the upload where the size can vary according to the total size of the file). [gsutil code][1] dealing with this.

[1]: https://github.com/GoogleCloudPlatform/gsutil/blob/ccca18e64f3f7d87c989675f8673af7bdecae0d1/gslib/util.py#L464

### PR DESCRIPTION
**Pull request by dhermes**
Fixes #546.

See #606 for old discussion. This was made possible by #811.

### CODE REVIEW

File: gcloud/storage/blob.py [hunk 1]
```diff
@@ -70,9 +72,35 @@ def __init__(self, name, bucket=None):
 
         super(Blob, self).__init__(name=name)
 
+        self._chunk_size = None  # Needs to be defined in __init__.
```
Comment (tseaver) [inline gcloud/storage/blob.py]: Shouldn't need to assign the backing attribute:  the property setter does it just fine.

Comment (dhermes) [inline gcloud/storage/blob.py]: Was a sacrifice to lint gods.

Comment (tseaver) [inline gcloud/storage/blob.py]: Could it be a class-level default instead?

Comment (dhermes) [inline gcloud/storage/blob.py]: DERP. Thanks.

**Comment by dhermes** [comment]
@tseaver I moved the default `_chunk_size` to class.

**Comment by tseaver** [comment]
LGTM

**Pull request merged**